src/skmultiflow/anomaly_detection/half_space_trees.py

```diff
@@ -0,0 +1,548 @@
+
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (n_samples, n_features)
+            Samples for which we want to predict the labels.
+
+        max_score: float
+            The maximum score of an instance could have in the tree.
+
+        Returns
+        -------
+        numpy.array
+            Predicted the probabilities of all the labels for all instances in X.
+
+        """
+        r, _ = get_dimensions(X)
+        predictions = []
+        for i in range(r):
+            votes = self.get_votes_for_instance(X[i], max_score).copy()
+            y_proba = np.zeros(int(max(votes.keys())) + 1)
+            for key, value in votes.items():
+                y_proba[int(key)] = value
+                predictions.append(y_proba)
+        return np.array(predictions)
+
+    def get_votes_for_instance(self, X, max_score):
+        """ Get class votes for a single instance.
+
+        Parameters
+        ----------
+        X: numpy.ndarray of length equal to the number of features.
+            Instance attributes.
+
+        max_score: float
+            The maximum score of an instance could have in the tree.
+        Returns
+        -------
+        dict (class_value, weight)
+
+        """
+        if self.root is not None and self.is_learning_phase_on is not True:
+            score = self.anomaly_score(X)
+            anomaly_score = 1 - score / max_score
+            return {0: 1 - anomaly_score, 1: anomaly_score}
+        else:
+            return {0: 0.5, 1: 0.5}
+
+    def anomaly_score(self, X):
+        """
+        Calculates anomaly score of a given instance X.
+
+        Parameters
+        ----------
+        X: numpy.ndarray of shape (1, n_features)
+            Instance attributes.
+
+        Returns
+        -------
+
+        """
+        return self._anomaly_score(self.root, X)
+
+    def _anomaly_score(self, node, X):
+        """
+        Private function to calculate anomaly score.
+
+        Parameters
+        ----------
+        node: HalfSpaceTreeNode
+            Current node
+
+        X: numpy.ndarray of shape (1, n_features)
+            Instance attributes.
+
+        Returns
+        -------
+        anomaly_score: float
+            Anomaly score of the instance X.
+
+        """
+        anomaly_score = 0.5
+        if node is not None:
+            if node.internal_node is not True or node.r < self.size_limit:
+                anomaly_score = node.r * pow(2.0, node.depth)
+            else:
+                if X[node.split_attribute] > node.split_value:
+                    anomaly_score = self._anomaly_score(node.right, X)
+                else:
+                    anomaly_score = self._anomaly_score(node.left, X)
+
+        return anomaly_score
+
+    def build_tree(self, min, max, current_depth=0):
+        """
+        Builds a single half space tree.
+
+        Parameters
+        ----------
+        min: Array of floats
```

Comment:
`min` and `max` names are shadowing the corresponding Python functions. Consider instead something like `min_values` and `max_values`